src/main/java/seedu/address/model/deliverymodel/DeliveryBook.java

```diff
@@ -1,14 +1,21 @@
 package seedu.address.model.deliverymodel;
 
 import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.core.Messages.MESSAGE_REDO_LIMIT_REACHED;
+import static seedu.address.commons.core.Messages.MESSAGE_UNDO_LIMIT_REACHED;
+import static seedu.address.model.Model.MODEL_STATE_LIMIT;
 
+import java.util.ArrayList;
 import java.util.List;
 
 import javafx.collections.ObservableList;
+import seedu.address.logic.commands.exceptions.UndoRedoLimitReachedException;
 import seedu.address.model.delivery.Delivery;
 import seedu.address.model.delivery.UniqueDeliveryList;
 
 public class DeliveryBook implements ReadOnlyDeliveryBook {
+    private static List<DeliveryBook> deliveryBookStateList = new ArrayList<>();
```

Comment:
Perhaps instantiate arraylist with a initial capacity to optimize performance (remember from 2040s they taught us that arraylist copies over and doubles in size everytime its limit is hit)

Reply:
Eg  private static List<DeliveryBook> deliveryBookStateList = new ArrayList<>(MODEL_STATE_LIMIT);

Reply:
and i feel that it shouldn't be a static variable simply because we may use different instantiation of deliverybooks in testing as well

Reply:
ok let me try

Reply:
issue with this is that it becomes hard to make a canRedo() method without adding another pointer, which makes things messy. I think our array limit will never be set beyond a few 1000 so it's not an issue (and if it is ever changed, the code can be refactored then)

Reply:
ArrayList.size() != Capacity btw @halcon-blanco

Reply:
oh sheez for some reason my github didn't show all your comments

Reply:
ok so for the static variable, I luckily saw the issue too and just fixed it. as for the capacity != size part, yeah my bad haha still rusty from 2040 times will add it in

Reply:
ok you can check it out now!